### ISSUE TITLE
How can I migrate Redis A db0 to redis B db2

### ISSUE THREAD

**Issue by calvin2020** [leonchen83/redis-rdb-cli#78]
Hi:
I have a question after reading the opration of  redis-rdb-cli,I want to migrate one Redis db0 to another Redis db2,Can I use redis-rdb-cli to solve my problem,look forward to geting your reply,
thanks

**Comment by leonchen83** [comment]
```
# backup source redis db0 to db2
rdt -b redis://srouce-ip:6379 -d 0 -o /path/to/dump.rdb --goal 2
# restore to target redis
rmt -s /path/to/dump.rdb -m redis://target-ip:6379 -r
```

**Comment by calvin2020** [comment]
> ```
> # backup source redis db0 to db2
> rdt -b redis://srouce-ip:6379 -d 0 -o /path/to/dump.rdb --goal 2
> # restore to target redis
> rmt -s /path/to/dump.rdb -m redis://target-ip:6379 -r
> ```
> 
> more details refer to [issue 23](https://github.com/leonchen83/redis-rdb-cli/issues/23)

**Comment by calvin2020** [comment]
> > ```
> > # backup source redis db0 to db2
> > rdt -b redis://srouce-ip:6379 -d 0 -o /path/to/dump.rdb --goal 2
> > # restore to target redis
> > rmt -s /path/to/dump.rdb -m redis://target-ip:6379 -r
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > more details refer to [issue 23](https://github.com/leonchen83/redis-rdb-cli/issues/23)

Thanks! I'll try it

**Comment by calvin2020** [comment]
![image](https://github.com/user-attachments/assets/16e918a1-7224-4ff3-8880-fb7ed22d3838)
I have tried according to your reply on same server(192.168.137.209),but failed to migrate redis(192.168.137.209) db0 to redis(192.168.137.207) db2,would you please help me to check it ,thanks

**Comment by leonchen83** [comment]
you shoud first convert that rdb to json format and pasted that json to here
```
rct -f jsonl -s /path/to/dump_db0.rdb -o /path/to/dump.jsonl
```

**Comment by calvin2020** [comment]
![image](https://github.com/user-attachments/assets/484a2e83-57a9-4967-ac31-c4d80f8f41d5)

the json files is empty,but my source redis db0 has keys
![image](https://github.com/user-attachments/assets/f5cf647b-a33b-4487-902c-acc0ef67744d)

**Comment by leonchen83** [comment]
wired, I'll test in my local redis

**Comment by calvin2020** [comment]
> wired, I'll test in my local redis

OK,if you find the root cause,please inform me,thanks

**Comment by leonchen83** [comment]
I knew the reason, this is a bug. I will fix it in next version

**Comment by leonchen83** [comment]
you could use following method to avoid this bug
```
rdt -b redis://source:6379  -d 0 -o /path/to/dump.rdb
rdt -b /path/to/dump.rdb -o /path/to/dump_db2.rdb -g 2
rmt -s /path/to/dump_db2.rdb -m redis://target:6379 -r
```

**Comment by calvin2020** [comment]
> you could use following method to avoid this bug
> 
> ```
> rdt -b redis://source:6379  -d 0 -o /path/to/dump.rdb
> rdt -b /path/to/dump.rdb -o /path/to/dump_db2.rdb -g 2
> rmt -s /path/to/dump_db2.rdb -m redis://target:6379 -r
> ```

Get it,I'll try it,really appreciate

**Comment by leonchen83** [comment]
Fixed in version 0.9.6. just use following method 
```
# backup source redis db0 to db2
rdt -b redis://srouce-ip:6379 -d 0 -o /path/to/dump.rdb --goal 2
# restore to target redis
rmt -s /path/to/dump.rdb -m redis://target-ip:6379 -r
```

**Comment by calvin2020** [comment]
> Fixed in version 0.9.6. just use following method
> 
> ```
> # backup source redis db0 to db2
> rdt -b redis://srouce-ip:6379 -d 0 -o /path/to/dump.rdb --goal 2
> # restore to target redis
> rmt -s /path/to/dump.rdb -m redis://target-ip:6379 -r
> ```

I just tested,it's OK now,thanks
